### ISSUE TITLE
The Future of "Button inside input"

### ISSUE THREAD

**Issue by mistergone** [cfpb/design-manual#441]
If you check out the current version of our `cf-forms` portion of Capital Framework, you will find our "Button Inside Input" style:

<img width="929" alt="screen shot 2016-09-16 at 12 03 06 pm" src="https://cloud.githubusercontent.com/assets/1490703/18592658/f9ebc752-7c05-11e6-9f0b-45e34c68f182.png">

But @KimberlyMunoz rightfully pointed out that the spacing of this element is very precarious, and translations may break it. I'm only fluent in English, so I won't take a stab at "Clear" in another language, but here's the Klingon terms for "remove words":

<img width="944" alt="screen shot 2016-09-16 at 12 09 11 pm" src="https://cloud.githubusercontent.com/assets/1490703/18592781/85f41362-7c06-11e6-9757-5bfb168ea899.png">

Translations are a problem, but also any text other than "Clear X" would break our current implementation - so "Clear text" would also break it.

While moving this code into atomic design, I attempted to make the button variable length. I was able to implement this, but the design broke when the input received focus. My attempts to fix the focus issue eventually ended with success, but only in a few browsers. After a lot of trying, I still wasn't able to make it work in Internet Explorer. (Some of my failures can be found in the discussion of the PR **[here](https://github.com/cfpb/capital-framework/pull/397)**).

For now, we've stopped trying to fix this problem for our next release. However, this problem will come back to us when we try internationalization for certain. So I'd like to propose the following:
- If someone knows of a working implementation of a design like this, we could review it and see if it would work for us. Our limited research has show working implementations like this design use JavaScript to shore everything up (which is not something we want to do).
- We consider removing the "Clear" text from the button and go simply with an "X" icon. The "X" icon is always a fixed width and thus removes the need for the button to be of a variable width.
- We consider moving the "Clear" button outside of the input. That is, the input could receive focus (the border and outline included) and other states, but the "Clear" button would exist outside of that input box. This should allow the "Clear" button to be a variable width.
- Someone proposes a solution aside from the ones I've suggested above. 😄 

I'm all too happy to answer questions, go over each solution I tried, and otherwise get or give feedback about the intractable problems we've encountered trying to implement this design in a way that allows the "Clear" text to be variable width!

**Comment by caheberer** [comment]
@mistergone where were you trying to implement this? I'm unfamiliar with us using this convention. Is a clear even necessary?

**Comment by jimmynotjim** [comment]
@mistergone didn't we end up resolving this during the move to v4?

**Comment by jimmynotjim** [comment]
@mistergone is this still a concern or can we close it?

@caheberer I should have mentioned, this is used in the search bar on cfgov-refresh and is documented in Capital Framework, but I can't find it in the Design Manual. It should probably be added under Form Fields, but I'm not sure

**Comment by nataliafitzgerald** [comment]
@jimmynotjim - Do we only use this for global search? 

Following up on @caheberer's comment - Do we think that clear is necessary?

**Comment by jimmynotjim** [comment]
@nataliafitzgerald Yes, as far as I know it's only been used for the global search.
